Purchases/Attribution/RCAttributionFetcher.m

```diff
@@ -164,22 +137,54 @@ - (void)postAttributionData:(NSDictionary *)data
     }
 }
 
-- (void)postAppleSearchAdsAttributionCollection {
+- (BOOL)isAuthorizedToPostSearchAds {
+#if APP_TRACKING_TRANSPARENCY_REQUIRED
+    if (@available(iOS 14, macos 11, tvos 14, *)) {
+        NSOperatingSystemVersion minimumOSVersionRequiringAuthorization = { .majorVersion = 14, .minorVersion = 5, .patchVersion = 0 };
+
+        BOOL needsTrackingAuthorization = [self.systemInfo isOperatingSystemAtLeastVersion:minimumOSVersionRequiringAuthorization];
+
+        Class<FakeATTrackingManager> _Nullable trackingManagerClass = [self.attributionFactory trackingManagerClass];
+        if (!trackingManagerClass && needsTrackingAuthorization) {
+            RCWarnLog(@"%@", RCStrings.attribution.search_ads_attribution_cancelled_missing_att_framework);
+            return NO;
+        }
+        NSInteger authorizationStatus = [trackingManagerClass trackingAuthorizationStatus];
+        BOOL authorized = (needsTrackingAuthorization && authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized)
```

Comment:
I think this is the same `authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized || (!needsTrackingAuthorization && authorizationStatus == FakeATTrackingManagerAuthorizationStatusNotDetermined)` and easier to read _in my opinion_. What do you think?

Reply:
very much agreed! thanks

Reply:
updated